test/test_euler.py

```diff
@@ -0,0 +1,781 @@
+                )
+            )
+            err = np.abs(err - p0)
+            assert err < tolerance
+            if err > maxmomerr:
+                maxmomerr = err
+
+        errrec = np.maximum(err1, maxmomerr)
+        eoc_rec1.add_data_point(1.0 / nel_1d, errrec)
+
+    message = (
+        f"standalone Errors:\n{eoc_rec0}"
+        f"boundary Errors:\n{eoc_rec1}"
+    )
+    logger.info(message)
+    assert (
+        eoc_rec0.order_estimate() >= order - 0.5
+        or eoc_rec0.max_error() < 1e-9
+    )
+    assert (
+        eoc_rec1.order_estimate() >= order - 0.5
+        or eoc_rec1.max_error() < 1e-9
+    )
+
+
+@pytest.mark.parametrize("dim", [1, 2, 3])
+@pytest.mark.parametrize("order", [1, 2, 3])
+def test_uniform_rhs(ctx_factory, dim, order):
+    """Tests the inviscid rhs using a trivial
+    constant/uniform state which should
+    yield rhs = 0 to FP.  The test is performed
+    for 1, 2, and 3 dimensions.
+    """
+    cl_ctx = ctx_factory()
+    queue = cl.CommandQueue(cl_ctx)
+    logger = logging.getLogger(__name__)
+
+    tolerance = 1e-9
+    maxxerr = 0.0
+    #            from pytools.convergence import EOCRecorder
+    #            eoc_rec0 = EOCRecorder()
+    #            eoc_rec1 = EOCRecorder()
+
+    for nel_1d in [4, 8, 12]:
+        from meshmode.mesh.generation import generate_regular_rect_mesh
+        mesh = generate_regular_rect_mesh(
+            a=(-0.5,) * dim, b=(0.5,) * dim, n=(nel_1d,) * dim
+        )
+
+        logger.info(
+            f"Number of {dim}d elements: {mesh.nelements}"
+        )
+
+        discr = EagerDGDiscretization(cl_ctx, mesh, order=order)
+
+        mass_input = discr.zeros(queue)
+        energy_input = discr.zeros(queue)
+
+        # this sets p = p0 = 1.0
+        mass_input[:] = 1.0
+        energy_input[:] = 2.5
+        mom_input = make_obj_array(
+            [discr.zeros(queue) for i in range(discr.dim)]
+        )
+        fields = flat_obj_array(
+            mass_input, energy_input, mom_input
+        )
+
+        expected_rhs = make_obj_array(
+            [discr.zeros(queue) for i in range(discr.dim + 2)]
+        )
+
+        boundaries = {BTAG_ALL: DummyBoundary()}
+        inviscid_rhs = inviscid_operator(discr, fields, boundaries)
+        rhs_resid = inviscid_rhs - expected_rhs
+
+        rho_resid = split_conserved(dim, rhs_resid).mass
+        rhoe_resid = split_conserved(dim, rhs_resid).energy
+        mom_resid = split_conserved(dim, rhs_resid).momentum
+
+        rho_rhs = split_conserved(dim, inviscid_rhs).mass
+        rhoe_rhs = split_conserved(dim, inviscid_rhs).energy
+        rhov_rhs = split_conserved(dim, inviscid_rhs).momentum
+
+        message = (
+            f"rho_rhs  = {rho_rhs}\n"
+            f"rhoe_rhs = {rhoe_rhs}\n"
+            f"rhov_rhs = {rhov_rhs}"
+        )
+        logger.info(message)
+
+        assert np.max(np.abs(rho_resid.get())) < tolerance
+        assert np.max(np.abs(rhoe_resid.get())) < tolerance
```

Comment:
Norms?

Reply:
norms are coming in with array-context port effort